### DISCUSSION TITLE
Lua for detecting emoji fonts in LaTeX

### DISCUSSION THREAD

**Discussion by avidseeker** [jgm/pandoc#10011] [Q&A]
Until automatic fallback handling by luatex (https://github.com/latex3/fontspec/issues/526, https://github.com/latex3/fontspec/issues/428) is implemented, I'm considering using pandoc as a workaround for specifying the Emoji font. Does pandoc have a representation for emojis in its AST? I expect it so since it has `emoji` markdown extension.

I'm aware of `*fontfallback` metadata keys in pandoc, but the mention of "Font fallbacks only work with lualatex", but that didn't make a change (I'm converting markdown --> latex).

The workaround I'm thinking of is: for every emoji `🤔` in markdown, it needs to be replaced with `\emoji{🤔}`. Where `\emoji` is a LaTeX macro for using the right font for emojis (see: [TeX SE](https://tex.stackexchange.com/questions/3695))

**Comment by badumont** [comment]
Another solution is described here: ***@***.***/T/#m8a94f188b30caecdf834c39f6427badb5bf2901d

(where \usepackage{newunicodefont} is to be replaced with \usepackage{newunicodechar})

**Reply by avidseeker** [reply]
You sent a censored URL?

**Reply by badumont** [reply]
Interesting, Github seems to redact URLs sent by email if they contain an @. Let's try again: https://inbox.vuxu.org/pandoc-discuss/CAFC_yuSQ+=nPiAC43sEG09K4UhpyuY2tHvjHqBe5ZxZ2A8NKsQ@mail.gmail.com/T/#m8a94f188b30caecdf834c39f6427badb5bf2901d

**Comment by jgm** [comment]
> Does pandoc have a representation for emojis in its AST?

If the emoji extension is enabled, then e.g. `:+1:` will be parsed as
```
[ Span
        ( "" , [ "emoji" ] , [ ( "data-emoji" , "+1" ) ] )
        [ Str "\128077" ] ]
```
But plain unicode emojis will just be regular text.
